src/lib/comms/sol-oic-security.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "sol-coap.h"
+#include "sol-oic-client.h"
+#include "sol-oic-common.h"
+#include "sol-str-slice.h"
+
+#define MACHINE_ID_LEN 16
+
+struct sol_oic_security;
+
+struct sol_oic_security *sol_oic_server_security_add(
+    struct sol_coap_server *server, struct sol_coap_server *server_dtls);
+void sol_oic_server_security_del(struct sol_oic_security *security);
+
+struct sol_oic_security *sol_oic_client_security_add(
+    struct sol_coap_server *server, struct sol_coap_server *server_dtls);
+void sol_oic_client_security_del(struct sol_oic_security *security);
+
+bool sol_oic_security_get_is_paired(const struct sol_oic_security *security,
+    struct sol_str_slice device_id);
+int sol_oic_security_pair_request(struct sol_oic_security *security,
+    struct sol_oic_resource *resource, enum sol_oic_pairing_method pm,
+    void (*paired_cb)(void *data, enum sol_oic_security_pair_result result), void *data);
+
+bool sol_oic_set_token_and_mid(struct sol_coap_packet *pkt, int64_t *token);
+const uint8_t *get_machine_id(void);
```

Comment:
On Tue, 01 Mar 2016 06:36:37 -0800, Otávio wrote:
> > +struct sol_oic_security *sol_oic_server_security_add(
> > +    struct sol_coap_server *server, struct sol_coap_server *server_dtls);
> > +void sol_oic_server_security_del(struct sol_oic_security *security);
> > +
> > +struct sol_oic_security *sol_oic_client_security_add(
> > +    struct sol_coap_server *server, struct sol_coap_server *server_dtls);
> > +void sol_oic_client_security_del(struct sol_oic_security *security);
> > +
> > +bool sol_oic_security_get_is_paired(const struct sol_oic_security *security,
> > +    struct sol_str_slice device_id);
> > +int sol_oic_security_pair_request(struct sol_oic_security *security,
> > +    struct sol_oic_resource *resource, enum sol_oic_pairing_method pm,
> > +    void (*paired_cb)(void *data, enum sol_oic_security_pair_result result), void *data);
> > +
> > +bool sol_oic_set_token_and_mid(struct sol_coap_packet *pkt, int64_t *token);
> > +const uint8_t *get_machine_id(void);
> 
> sol_platform_get_machine_id is encoded as hex and get_machine_id is a 16 byte array with decoded bytes from platform id.

In which case, I'd rather have that function added next to the other. I
can imagine it being useful for other scenarios too.